src/model/block/ISetGroupMembershipBlockConfig.ts

```diff
@@ -1,23 +1,25 @@
 import {IBlockConfig} from '../..'
 
-export interface ISetGroupMembershipBlockConfig extends IBlockConfig {
-  /**
-   * An identifier for the group that membership will be set within.
-   */
+interface IGroup {
   group_key: string
-  /**
-   * A human-readable label in addition to the group_key, in cases where the group_name needs to be displayed to the Contact.
-   */
-  group_name?: string
-  /**
-   * Determines the membership state:
-   * - falsy to remove the contact from the group,
-   * - truthy to add,
-   * - and null for no change to the existing membership.
-   *
-   * Its type is string because it may contain expressions too.
-   *
-   * @format floip-expression
-   */
-  is_member: string
+  group_name: string
 }
+
+/**
+ * Adding or removing contact from group(s)
+ */
+interface IEditGroupMembership {
+  is_member: boolean
+  groups: IGroup[]
+}
+
+/**
+ * Removing contact from all groups
+ */
+interface IClearGroupMembership {
+  clear: true
+}
+
+type GroupMembershipSetting = IEditGroupMembership | IClearGroupMembership
+
+export type ISetGroupMembershipBlockConfig = IBlockConfig & GroupMembershipSetting
```

Comment:
Alternatively, the two interfaces could extend IBlockConfig which would remove the need for GroupMembershipSetting (one line less).
Found here: https://stackoverflow.com/a/40510700/11980366